source/Octopus.TestPortForwarder/SocketPump.cs

```diff
@@ -69,15 +91,34 @@ async Task PausePump(CancellationToken cancellationToken)
                 cancellationToken.ThrowIfCancellationRequested();
             }
         }
-
-        static async Task WriteToSocket(Socket writeTo, byte[] inputBuffer, int totalBytesToSend, CancellationToken cancellationToken)
+        
+        async Task WriteToSocketDelayingSendingTheLastNBytes(Socket writeTo, byte[] buffer, int bufferLength, int delaySendingLastNBytes, CancellationToken cancellationToken)
+        {
+            int howMuchToSend = bufferLength - delaySendingLastNBytes;
```

Comment:
Nit: Why aren't we using 'vars' in this file? If we can, I think we should 😁

Reply:
Not sure